File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void view(TreeNode* root, vector<int>&v, int level){
        if(root == NULL){
            return;
        }
        if(v.size() == level){
            v.push_back(root->val);
        }
        view(root->right,v,level+1);
        view(root->left,v,level+1);
    }
    vector<int> rightSideView(TreeNode* root) {
        vector<int>v;
        view(root,v,0); 
        return v;
    }
};

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        ArrayList<Integer> arr = new ArrayList<>();
        Queue<TreeNode> q = new ArrayDeque<>();
        if(root == null)return arr;
        q.add(root);
        while(q.size() > 0){
            arr.add(q.peek().val);
            int size = q.size();
            for(int i = 0 ;i < size; i++){
                TreeNode node = q.remove();
                if(node.right != null)q.add(node.right);
                if(node.left != null)q.add(node.left);
            }
        }
        return arr;
    }
}